Updated data processing and added initial details dashboard.

ngillux opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a [descriptive title](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request?id=prepare-a-good-pr-title) (**not** a vague title like `Update index.md`)

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->



<!-- Feel free to add any additional description of changes below this line -->

utsab commented:
@ngillux After merging PR #473, the current PR is showing merge conflicts in two files.  Can you take a look at the merge conflicts and resolve them?

ngillux commented:
@utsab I resolved the merge conflict - I had changed a property name, and removed a redundant property in an object in this PR. You should be able to see/test the updated changes in a Codespace now. Let me know if there's any other issues.

Komal914 commented on pages/dashboard/v2/details/[id]/[studentEmail].js:
```diff
@@ -51,19 +60,47 @@ export async function getServerSideProps(context) {
+  let dashboardObjs = createDashboardObject(superBlockJsons);
+
+  // console.log('DDASHBOARD OBJS ===>', dashboardObjs)
+  let studentData = await getIndividualStudentData(studentEmail);
```
ISR should be used for the student data API response. This will allow us to avoid merging the data, and simply update the respective UI elements.

Komal914 reviewed (commented):
Let's avoid merging superblocks with student progress blocks, and just update the UI. Should increase optimization and maintain the functionality.

Komal914 commented on components/details/DetailsBoard.js:
```diff
@@ -0,0 +1,64 @@
+  let mergedDashboardData = [];
+
+  // reshape data to pass into UI
+  allCurriculumData.forEach(superblockData => {
```
lines 13-35: Let's remove this  functionality to API processor an a function we utilize inside this file. Something like: updatedAllStudentCurriculumData(email). Then we can call this function and add the arguement for email inside this file.

Komal914 commented on pages/dashboard/v2/details/[id]/[studentEmail].js:
```diff
@@ -82,9 +119,17 @@ export default function StudentDetails({
+          </div>
+
+          <DetailsBoard
+            studentCurriculumData={studentCurriculumData}
```
we no longer need to pass props here since this is being imported from API processor

ngillux replied:
We will still need props here because [getServerSideProps() ](https://nextjs.org/docs/pages/building-your-application/data-fetching/get-server-side-props) is what gives us access to data stored on the server, which is needed as the reference/parameters for the utility functions in api_processor.js. Even though functions from api_processr.js can be called in the other components, it is still necessary to fetch the data on the server first - and we need getServerSideProps() + context.params for this.
